module/friendship/adapter/postgres/repository/subscription.go

```diff
@@ -40,24 +40,29 @@ func (f SubscriptionRepository) UpdateStatus(ctx context.Context, id string, sta
 		ID:     id,
 		Status: int(status),
 	}
-	_, err := m.Update(ctx, f.db.Model(ctx), boil.Whitelist(model.SubscriptionColumns.Status, model.SubscriptionColumns.UpdatedAt))
+	effectedRows, err := m.Update(ctx, f.db.Model(ctx), boil.Whitelist(model.SubscriptionColumns.Status, model.SubscriptionColumns.UpdatedAt))
 	if err != nil {
 		return common.ErrDB(err)
 	}
+
+	if effectedRows != 1 {
+		return common.ErrDB(domain.ErrUpdateRecordNotFound)
+	}
+
 	return nil
 }
 
-func (f SubscriptionRepository) UnsertSubscription(ctx context.Context, sub domain.Subscription) error {
+func (f SubscriptionRepository) UnsertSubscription(ctx context.Context, sub domain.Subscription) (string, error) {
 	m := convert.ToSubscriptionModel(sub)
 	if m.ID == "" {
 		m.ID = util.GenUUID()
 	}
 	conflictFields := []string{model.SubscriptionColumns.UserID, model.SubscriptionColumns.SubscriberID}
 	err := m.Upsert(ctx, f.db.Model(ctx), true, conflictFields, boil.Whitelist(model.SubscriptionColumns.Status, model.FriendshipColumns.UpdatedAt), boil.Infer())
 	if err != nil {
-		return common.ErrDB(err)
+		return "", common.ErrDB(err)
 	}
-	return nil
+	return m.ID, nil
```

Comment:
What's the purpose of returning the ID here ?

Reply:
I am using this ID to check in unit test